src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -2222,29 +2222,30 @@
 .rs.addFunction("parsePackageDependencies", function(contents, extension)
 {
    # NOTE: the following regular expressions were extracted from knitr;
-   # we pull these out here just to avoid potentially loading the knitr
-   # package without the user's consent
+   # They were modified to search only R code chunks and skip other chunks
+   # such as python, bash, sql, etc that do not contain R dependencies
+   # Inline R code in markdown docs is also skipped
    code <- if (identical(extension, ".R"))
       contents
    else if (identical(extension, ".Rmd"))
       .rs.extractRCode(contents,
-                       "^[\t >]*```+\\s*\\{([a-zA-Z0-9_]+.*)\\}\\s*$",
+                       "^[\t >]*```+\\s*\\{([Rr]+.*)\\}\\s*$",
                        "^[\t >]*```+\\s*$")
    else if (identical(extension, ".Rnw"))
       .rs.extractRCode(contents,
                        "^\\s*<<(.*)>>=.*$",
                        "^\\s*@\\s*(%+.*|)$")
    else if (identical(extension, ".qmd"))
      .rs.extractRCode(contents,
-                      "^[\t >]*```+\\s*\\{([a-zA-Z0-9_]+.*)\\}\\s*$",
+                      "^[\t >]*```+\\s*\\{([Rr]+.*)\\}\\s*$",
```

Comment:
Same as above?

Reply:
```suggestion
                      "^[\t >]*```+\\s*\\{[Rr]\\b+.*\\}\\s*$",
```